src/simulate.jl

```diff
@@ -120,25 +120,31 @@ end
 Returns a simulation step function for the given `dt1` with stochastic elements, introducing possible events like vehicular needs and delayed returns.
 The function chooses between `simustep_alldrive`, `simustep_allplug`, or `simustep_event` probabilistically.
 """
-function get_simustep_stochastic(dt1::DateTime)
+function get_simustep_stochastic(dt1::DateTime, drive_starts_time, park_starts_time)
     date_part = Dates.Date(dt1)
-    dt_9am = DateTime(date_part, Dates.Time(9, 0, 0))
-    dt_5pm = DateTime(date_part, Dates.Time(17, 0, 0))
+    dt_9am = DateTime(date_part, drive_starts_time)
+    dt_5pm = DateTime(date_part, park_starts_time)
 
-    if dt_5pm - periodstep(1) ≤ dt1 < dt_5pm && rand() < prob_delayed_return
-        return simustep_base
+    rand_delayed_return = rand()
+    rand_event_return = rand()
+    rand_event = rand()
+
+    if dt_5pm - periodstep(1) ≤ dt1 < dt_5pm && rand_delayed_return < prob_delayed_return
+        delayed_return = 1 ## the event caused a later return than anticipated?
+        return simustep_base, (delayed_return, 0, 0)
     end
-    if rand() < prob_event_return
+    if rand_event_return < prob_event_return
+        event_return = 1 ## the event caused a return 
         if dt_9am - periodstep(1) ≤ dt1 < dt_5pm - periodstep(1)
-            return simustep_alldrive
+            return simustep_alldrive, (0, event_return, 0)
         else
-            return simustep_allplug
+            return simustep_allplug, (0, event_return, 0)
         end
     end
-    if rand() < prob_event
-        vehicles_needed = sample(1:vehicles, Weights(prob_event_vehicles))
-        return (vehicles_plugged_1::Float64, vehicles_avail_1::Float64, soc_avail_1::Float64, soc_driving_1::Float64) -> simustep_event(vehicles_needed, vehicles_plugged_1, vehicles_avail_1, soc_avail_1, soc_driving_1)
+    if rand_event < prob_event
+        vehicles_needed = sample(1:vehicles, Weights(prob_event_vehicles)) ## the event caused a vehicle event?
```

Comment:
I was thinking about the Delmarva case for this. Imagine that you have a bad storm (e.g., with downed powerlines), and that forces some number of vehicles to have to start driving.